doc/tools/docgram/README.md

```diff
@@ -1,123 +1,142 @@
-`productionlist` displays.  Updates to `tacn` and `cmd` must be made
+`doc_grammar` extracts Coq's grammar from `.mlg` files, edits it and inserts it in
+chunks into `.rst` files.  The tool currently inserts Sphinx
+`productionlist` constructs.  It also generates a file with `prodn` constructs
+for the entire grammar, but updates to `tacn` and `cmd` must be done
 manually since the grammar doesn't have names for them as it does for
 nonterminals.  There is an option to report which `tacn` and `cmd` were not
-found in the .rst files.  `tacv` and `cmdv` are not processed at all.
+found in the `.rst` files.  `tacv` and `cmdv` constructs are not processed at all.
 
 The mlg grammars present several challenges to generating an accurate grammar
 for documentation purposes:
+
 * The 30+ mlg files don't define an overall order in which nonterminals should
-appear in a complete grammar.
+  appear in a complete grammar.
+
 * Even within a single mlg file, grammar rules and productions are often given
-in an order that's much different from what a reader of the documentation would
-expect.  In a small number of cases, the order used is necessary for technical
-reasons.
+  in an order that's much different from what a reader of the documentation would
+  expect.  In a small number of cases, changing the order in the mlg would change
+  how some inputs are parsed, in particular when the order determines how to distinguish
+  otherwise ambiguous inputs.
+
+  Strictly speaking, that means our grammar is not a context free grammar even though
+  we gloss over that distinction in the documentation.
+
 * For a few nonterminals, some productions are only available if certain plugins
-are activated (e.g. SSR).  Readers should be informed about these.
+  are activated (e.g. SSR).  Readers should be informed about these.
+
 * Some limited parts of the grammar are defined in OCaml, including lookahead symbols
-like `test_bracket_ident` and references to nonterminals in other files using
-qualified names such as `Prim.ident`.  A few symbols are defined multiple times,
-such as `scope` and `orient`.
+  like `test_bracket_ident` and references to nonterminals in other files using
+  qualified names such as `Prim.ident`.  A few symbols are defined multiple times,
+  such as `scope` and `orient`.
 
 What the tool does
 ------------------
-1. The tool reads all the mlg files and generates `fullGrammar`, which includes
+
+1. The tool reads all the `mlg` files and generates `fullGrammar`, which includes
 all the grammar without the actions for each production or the OCaml code.  This
 file is provided as a convenience to make it easier to examine the (mostly)
 unprocessed grammar of the mlg files with less clutter.  Nonterminals that use
 levels ("5" RIGHTA below) are modified, for example:
 
-```
-  tactic_expr:
-    [ "5" RIGHTA
-      [ te = binder_tactic -> { te } ]
-```
+    tactic_expr:
+      [ "5" RIGHTA
+        [ te = binder_tactic -> { te } ]
```

Comment:
I am not sure what you intended to do while removing the block quotes. You can keep them with the same indentation as within the numbered-point, and add no further indentation within.